src/Microsoft.OData.Edm/Schema/EdmModelBase.cs

```diff
@@ -173,25 +199,39 @@ public virtual IEnumerable<IEdmOperation> FindDeclaredBoundOperations(IEdmType b
         /// </returns>
         public virtual IEnumerable<IEdmOperation> FindDeclaredBoundOperations(string qualifiedName, IEdmType bindingType)
         {
-            IEnumerable<IEdmOperation> enumerable = this.FindDeclaredOperations(qualifiedName);
+            IEnumerable<IEdmOperation> enumerable = this.FindDeclaredBoundOperations(bindingType);  
+            
+            if(enumerable == null)
+            {
+                return new IEdmOperation[] { };
+            }
+
+            IList<IEdmOperation> matchedOperations = new List<IEdmOperation>();

```

Comment:
In hot path, something that can also have some performance impact (CPU and memory/GC), is to provide the initial capacity in the List constructor, when possible/available of course.

The default constructor initializes with Capacity of 4 and it doubles the size (which requires allocation and copy) every time needed when Add is invoked.

Even if we just know the MAX capacity expected and it is not crazy high, that is something worth providing because then we avoid all allocation and copies that come with adding elements to the list.

Reply:
An initial guess of 4 is probably a pretty good guess.

---
In reply to: [493170697](https://github.com/OData/odata.net/pull/1888#discussion_r493170697) [](ancestors = 493170697)